IPython/core/tests/test_interactiveshell.py

```diff
@@ -543,12 +542,10 @@ class TestSafeExecfileNonAsciiPath(unittest.TestCase):
 
     @onlyif_unicode_paths
     def setUp(self):
-        self.BASETESTDIR = tempfile.mkdtemp()
-        self.TESTDIR = join(self.BASETESTDIR, u"åäö")
+        self.BASETESTDIR = Path(tempfile.mkdtemp())
+        self.TESTDIR = self.BASETESTDIR / "åäö"
         os.mkdir(self.TESTDIR)
-        with open(
-            join(self.TESTDIR, "åäötestscript.py"), "w", encoding="utf-8"
-        ) as sfile:
+        with open(self.TESTDIR / "åäötestscript.py", "w", encoding="utf-8") as sfile:
```

Comment:
`Path` objects have a method for writing text: `path.write_text("text", encoding="utf-8")`

Reply:
You're right, didn't even think of that. I don't know what causes the `FileNotFoundError` though